util/util.go

```diff
@@ -152,7 +152,7 @@ func IsNodeReady(node *v1.Node) bool {
 	return false
 }
 
-// GetClusterFromMetadata returns the Cluster object (if present) using the object metadata.
+// GetClusterFromMetadata returns the Cluster object (if present) using the object metadata. 
```

Comment:
This method isn't used anymore. Should we remove it?

Reply:
i think it's OK to remove it as the method is quite simple. :)

Reply:
We should keep it in case providers are using it, or add it to the v1alpha3 provider migration guide

Reply:
We can mark it as deprecated for v1alpha3 and remove it in v1alpha4 (or whatever the following release is). It will help with conversion from v1alpha2 -> v1alpha3 as well.